baggage/baggage.go

```diff
@@ -358,10 +398,13 @@ func (m Member) Properties() []Property { return m.properties.Copy() }
 // String encodes Member into a header string compliant with the W3C Baggage
 // specification.
 func (m Member) String() string {
-	// A key is just an ASCII string. A value is restricted to be
-	// US-ASCII characters excluding CTLs, whitespace,
-	// DQUOTE, comma, semicolon, and backslash.
-	s := m.key + keyValueDelimiter + valueEscape(m.value)
+	//  W3C Baggage specification does not allow percent-encoded keys.
+	if !validateKey(m.key) {
+		return ""
+	}
```

Comment:
Is this needed?

Reply:
Yes, as the W3C does not allow percent-encoded keys, and this method is used in the W3C propagator.

https://github.com/open-telemetry/opentelemetry-go/pull/5132/files#diff-1b0d6480f6c659a565000afedab942883f1a1681893bdf522fa0992d1a48467aR732